### ISSUE TITLE
getCollections() won't get the collections

### ISSUE THREAD

**Issue by fdplacido** [key4hep/k4FWCore#60]
After trying to get some collections using the PodioDataSvc, read by PodioInput, I found that calling `getCollections()` won't return the collections. 
I opened this PR to fix it: https://github.com/key4hep/k4FWCore/pull/56 but that breaks the tests.
I figured out that I need to call `getReadCollections()`, but I still find quite confusing that `getCollections()` won't get the collections, but the data member, which then doesn't have the collections.

This should be either better documented or refactored to better indicate what the methods actually do.

**Comment by fdplacido** [comment]
Small example:

```cpp
PodioDataSvc* m_podioDataSvc = dynamic_cast<PodioDataSvc*>( evtSvc().get());
auto collections = m_podioDataSvc->getCollections();
# This will be 0
auto coll_size = collections.size();

auto collections2 = m_podioDataSvc->getReadCollections();
# This will be X > 0
auto coll_size2 = collections2.size();
```

**Comment by hegner** [comment]
I do not see any reason for why to have the 'getReadCollections' as public interface. So looks like a bug to me.

**Comment by tmadlener** [comment]
This should be fixed with #70

**Comment by andresailer** [comment]
I hope this is fixed as stated above
